tests/test_bad_config.rs

```diff
@@ -227,7 +227,10 @@ test!(bad_git_dependency {
         [dependencies]
         foo = { git = "file:.." }
     "#)
-    .file("src/lib.rs", "");
+    .file("src/lib.rs", "").file(".cargo/config", r#"
+    [net]
+    retry=0
```

Comment:
In theory these modifications to tests shouldn't be necessary. All of these are indicative of errors that can *never* be resolved, so they shouldn't print out warnings anyway